### PR TITLE
MULE-13792: Soap Service: add flag to log request/response

### PR DESCRIPTION
**Pull request by juandesi**

### CODE REVIEW

File: modules/soap-api/src/main/java/org/mule/runtime/soap/api/message/dispatcher/DefaultHttpMessageDispatcher.java [hunk 1]
```diff
@@ -48,25 +57,40 @@ public DefaultHttpMessageDispatcher(HttpClient client) {
...
       throw new DispatchingException("An error occurred while sending the SOAP request");
     } catch (TimeoutException e) {
       throw new DispatchingException("The SOAP request timed out", e);
     }
   }
 
+  /**
+   * Logs the content if it's log enabled.
+   */
+  private InputStream retrieveLogging(String title, InputStream content) {
```
Comment (alegmarra) [inline modules/soap-api/src/main/java/org/mule/runtime/soap/api/message/dispatcher/DefaultHttpMessageDispatcher.java]: why `retrieve`? this is not returning something different. Just `log` or `doLog` is much clearer

File: modules/soap-api/src/main/java/org/mule/runtime/soap/api/message/dispatcher/DefaultHttpMessageDispatcher.java [hunk 2]
```diff
@@ -48,25 +57,40 @@ public DefaultHttpMessageDispatcher(HttpClient client) {
...
+    if (LOGGER.isDebugEnabled()) {
+      String c = IOUtils.toString(content);
+      LOGGER.debug("Logging " + title);
```
Comment (marianogonzalez) [inline modules/soap-api/src/main/java/org/mule/runtime/soap/api/message/dispatcher/DefaultHttpMessageDispatcher.java]: Logging doesn't mean anything. Add more context to identify the request. THink of a highly concurrent scenario, you'll need to trace each message to an event or request id

File: modules/soap-api/src/main/java/org/mule/runtime/soap/api/message/dispatcher/DefaultHttpMessageDispatcher.java (hunk 1 above)
Comment (juandesi) [reply modules/soap-api/src/main/java/org/mule/runtime/soap/api/message/dispatcher/DefaultHttpMessageDispatcher.java]: if we consume the Stream while logging we need to return a new one.

File: modules/soap-api/src/main/java/org/mule/runtime/soap/api/message/dispatcher/DefaultHttpMessageDispatcher.java (hunk 2 above)
Comment (juandesi) [reply modules/soap-api/src/main/java/org/mule/runtime/soap/api/message/dispatcher/DefaultHttpMessageDispatcher.java]: makes sense, let me find a way to identify requests

Comment (alegmarra) [reply modules/soap-api/src/main/java/org/mule/runtime/soap/api/message/dispatcher/DefaultHttpMessageDispatcher.java]: does the context headers have some kind of correlationId?

File: modules/soap-api/src/main/java/org/mule/runtime/soap/api/message/dispatcher/DefaultHttpMessageDispatcher.java (hunk 1 above)
Comment (alegmarra) [reply modules/soap-api/src/main/java/org/mule/runtime/soap/api/message/dispatcher/DefaultHttpMessageDispatcher.java]: is still the same content, you just read it for logging

**Pull request merged**